src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -70,7 +69,37 @@ internal virtual void GetPolicies(JsonSerializerOptions options)
         {
             if (RuntimePropertyType.IsArray)
             {
-                EnumerableConverter = s_jsonEnumerableConverter;
+                EnumerableConverter = new DefaultArrayConverter();
+            }
+            else if (typeof(IEnumerable).IsAssignableFrom(RuntimePropertyType))
+            {
+                Type elementType = JsonClassInfo.GetElementType(RuntimePropertyType);
+
+                if (elementType == null)
+                {
+                    return;
+                }
+
+                if (RuntimePropertyType == typeof(IEnumerable<>).MakeGenericType(elementType))
```

Comment:
Could this inspect the generic type definition and the first generic argument instead?

`MakeGenericType` is going to do a bunch of type loading, only to do a comparison with the result and say "nope".

Reply:
This has been removed, but I still use a similar construct to determine if the `RuntimePropertyType` is implemented by `JsonEnumerableT<>`. Please advise if there's a better way.